ui_trt.py

```diff
@@ -78,61 +66,93 @@ def on_ui_tabs():
+                    #         opt_bs = gr.Slider(minimum=1, maximum=16, step=1, label="Optimal batch size", value=1, elem_id="trt_opt_bs")
+                    #         max_bs = gr.Slider(minimum=1, maximum=16, step=1, label="Maximum batch size", value=1, elem_id="trt_max_bs")
+
+                    #     with gr.Column(elem_id="trt_token_count"):
+                    #         min_token_count = gr.Slider(minimum=75, maximum=750, step=75, label="Minimum prompt token count", value=75, elem_id="trt_min_token_count")
+                    #         opt_token_count = gr.Slider(minimum=75, maximum=750, step=75, label="Optimal prompt token count", value=75, elem_id="trt_opt_token_count")
+                    #         max_token_count = gr.Slider(minimum=75, maximum=750, step=75, label="Maximum prompt token count", value=75, elem_id="trt_max_token_count")
+
+                    #     trt_extra_args = gr.Textbox(label='Extra arguments', value="", elem_id="trt_extra_args", info="Extra arguments for trtexec command in plain text form")
+
+                    #     with FormRow(elem_classes="checkboxes-row", variant="compact"):
+                    #         use_fp16 = gr.Checkbox(label='Use half floats', value=True, elem_id="trt_fp16")
+
+                    #     button_export_trt = gr.Button(value="Convert ONNX to TensorRT", variant='primary', elem_id="trt_convert_from_onnx")
 
             with gr.Column(variant='panel'):
                 trt_result = gr.Label(elem_id="trt_result", value="", show_label=False)
                 trt_info = gr.HTML(elem_id="trt_info", value="")
 
+
+        refresh_onnx_db.click(load_onnx_list,
+                              outputs=trt_source_filename_preset)
         button_export_unet.click(
             wrap_gradio_gpu_call(export_unet_to_onnx, extra_outputs=["Conversion failed"]),
             inputs=[onnx_filename, onnx_opset],
             outputs=[trt_result, trt_info],
         )
 
-        button_export_trt.click(
-            wrap_gradio_gpu_call(convert_onnx_to_trt, extra_outputs=[""]),
-            inputs=[trt_filename, trt_source_filename, min_bs, max_bs, min_token_count, max_token_count, min_width, max_width, min_height, max_height, use_fp16, trt_extra_args],
-            outputs=[trt_result, trt_info],
-        )
-
-        button_show_trt_command.click(
-            wrap_gradio_gpu_call(get_trt_command, extra_outputs=[""]),
-            inputs=[trt_filename, trt_source_filename, min_bs, max_bs, min_token_count, max_token_count, min_width, max_width, min_height, max_height, use_fp16, trt_extra_args],
-            outputs=[trt_result, trt_info],
+        button_export_preset_trt.click(
+            wrap_gradio_gpu_call(convert_onnx_to_trt_preset, extra_outputs=[""]),
+            inputs=[trt_filename, trt_source_filename_preset, profile_512_512_1, profile_512_512_2, profile_512_512_4, profile_768x768x1,profile_768x768x2, profile_768x768x4, use_fp16],
+            outputs=[trt_result, trt_info],      

```

Comment:
As said on the other function I don't like this bool stuff. Why don't we just send the element ID's of the enabled buttons as list ? From that you can infer the shape info.